File: .vscode/Python/Python_basic/Pattern_1.py
n=4

for i in range(1,n+1):
    num=1
    for j in range(1,(2*n)):
        if j>=5-i and j<=i+3:
            print(num,end=" ")
            num+=1 if j<n else -1
        else:
            print("  ",end="")
    print()

# i=1 j>=5-i <=i+3
# i=2 j>=3 <=5
# i=3 j>=2 <=6

File: .vscode/C/Week_4/Day_2/pattern_1.c
#include <stdio.h>

int main(){
    for(int i=0;i<9;i++){
        int num=1;
        for(int j=0;j<9;j++){
            if(i>=5){
                if(j>=i-4 && j<=12-i){
                    printf("%d ",num);
                    j>3?num--:num++;
                }
                else
                    printf("  ");
            }
            else{
                if(j>=4-i && j<=4+i){
                    printf("%d ",num);
                    j>3?num--:num++;
                }
                else
                    printf("  ");
            }
        }
        printf("\n");
    }
}


// i=0  j>=4-i or <=4+i
// i=1  j>=3 or <=5
// i=2  j=2,3,4,5,6
// 1=3  j=1,2,3,4,5,6,7
// i=4  j=0,1,2,3,4,5,6,7,8


// i=5 j=1,2,3,4,5,6,7
// i=6 j=2,3,4,5,6
// i=7 j>=i-4 <=12-i
// i=8 j=4
